subsys/net/ip/l2/ieee802154/ieee802154.c

```diff
@@ -233,7 +234,8 @@ static enum net_verdict ieee802154_recv(struct net_if *iface,
 	}
 
 	if (mpdu.mhr.fs->fc.frame_type == IEEE802154_FRAME_TYPE_BEACON) {
-		return ieee802154_handle_beacon(iface, &mpdu);
+		return ieee802154_handle_beacon(iface, &mpdu,
+						net_pkt_ieee802154_lqi(pkt));
```

Comment:
This would be cleaner if we just passed the pkt to handle_beacon instead of the mpdu and let ieee802154_handle_beacon() extract the lqi from the net_pkt directly.

Reply:
Yep, will do it in another patch though

Reply:
Actually I am revising the idea of changing this, it is better as it is right now where handle_beacon gets the pre-parsed mpdu and the lqi as parames instead of getting the net_pkt where it would require to re-parse all over again the 15.4 frame.

Reply:
your perogative. Just seems cleaner to pass the net_pkt instead of the mpdu which would then carry the packet values.